rmpv/src/ext/de.rs

```diff
@@ -534,6 +596,98 @@ impl<'de> Deserializer<'de> for &'de ValueRef<'de> {
     }
 }
 
+enum ExtData<'de> {
+    Owned(Vec<u8>),
+    Borrowed(&'de [u8])
+}
+
+struct ExtDeserializer<'de> {
+    tag: Option<i8>,
+    data: Option<ExtData<'de>>,
+}
+
+impl<'de> ExtDeserializer<'de> {
+    fn new_owned(tag: i8, data: Vec<u8>) -> Self {
+        ExtDeserializer {
+            tag: Some(tag),
+            data: Some(ExtData::Owned(data)),
+        }
+    }
+
+    fn new_ref(tag: i8, data: &'de [u8]) -> Self {
+        ExtDeserializer {
+            tag: Some(tag),
+            data: Some(ExtData::Borrowed(data)),
+        }
+    }
+}
+
+impl<'de> SeqAccess<'de> for ExtDeserializer<'de> {
+    type Error = Error;
+
+    fn next_element_seed<T>(&mut self, seed: T) -> Result<Option<T::Value>, Error>
+    where
+        T: DeserializeSeed<'de>,
+    {
+        if self.tag.is_some() || self.data.is_some() {
+            return Ok(Some(seed.deserialize(self)?));
+        }
+
+        Ok(None)
+    }
+}
+
+impl<'a, 'de: 'a> Deserializer<'de> for ExtDeserializer<'de> {
+    type Error = Error;
+
+    #[inline]
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+        where V: Visitor<'de>
+    {
+        Err(de::Error::invalid_type(de::Unexpected::Other("non tuple struct"), &"Ext tuple struct"))
```

Comment:
Some code, like that backing `#[serde(untagged)]` on enums will call `deserialize_any` and accept any result. I believe it's expected that `deserialize_any` will just call the appropriate visitor method, if it has type information.   

Since we know that we want to deserialize to a tuple struct, what would you think of replacing this with a call to `visitor.visit_seq()`? If the `Deserializer` code can't cope with a sequence, it'll still error, just slightly later.